index.js

```diff
@@ -152,7 +152,7 @@ export const globby = async (patterns, options = {}) => {
 	const [filter, tasks] = await Promise.all([getFilter(options), getTasks()]);
 	const paths = await Promise.all(tasks.map(task => fastGlob(task.pattern, task.options)));
 
-	return arrayUnion(...paths).filter(path_ => !filter(getPathString(path_)));
```

Comment:
`filter` already handles `objectMode` result, not needed here, `globbySync` and `globbyStream` also not doing this.